util/helm/creds.go

```diff
@@ -146,11 +147,30 @@ func (creds AzureWorkloadIdentityCreds) GetAccessToken() (string, error) {
 		return "", fmt.Errorf("failed to get Azure access token after challenge: %w", err)
 	}
 
-	// Access token has a lifetime of 3 hours
-	storeAzureToken(key, token, 2*time.Hour)
+	tokenExpiry, err := getJWTExpiry(token)
+	if err != nil {
+		tokenExpiry = time.Now()
```

Comment:
Shouldn't we just return up the error and bail? By hiding the error, we could end up in a situation where we're constantly requesting tokens and have no knowledge that the expiry logic is failing.